packages/common.vm/tools/vm.common/vm.common.psm1

```diff
@@ -1244,3 +1244,62 @@ public class Shell {
         VM-Write-Log-Exception $_
     }
 }
+
+
+# Usage example:
+# VM-Remove-DesktopFilesExcept -excludeFolders "PS_Transcripts", "Tools", "fakenet_logs" -excludeFiles "example.txt", "important.doc"
+function VM-Remove-DesktopFilesExcept {
+    param (
+        [Parameter(Mandatory=$false)]
+        [string[]]$excludeFolders,
+        [Parameter(Mandatory=$false)]
+        [string[]]$excludeFiles
+    )
+    $excludeFolders = $excludeFolders
```

Comment:
I think we should ensure the following folders are never deleted by adding them here to the provided argument:
- the tools directory (use env variable instead of hardcoded value)
- `fakenet_logs`

Reply:
I will add "PS_Transcripts" to this as well due an issue that can occur based on timing of execution of this function. 

If a restart happens and a PS log in this folder is generated and then this function is run, it will fail out due to a file being in use. 

I wasn't able to get it to cleanly bypass it for the moment and the circumstances are likely to be rare, but I think adding this for now won't be an issue. We can look into other ways to mitigate this in the future if needed.